File: src/test/java/io/binac/leetcode/SimplifyPathTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SimplifyPathTest {
    private final SimplifyPath.Solution1 solution1 = new SimplifyPath.Solution1();

    @Test
    void test1() {
        assertEquals("/home", solution1.simplifyPath("/home/"));
        assertEquals("/c", solution1.simplifyPath("/a/./b/../../c/"));
        assertEquals("/", solution1.simplifyPath("/../"));
        assertEquals("/home/foo", solution1.simplifyPath("/home//foo/"));
    }
}

File: cpp/tests/SimplifyPathTest.cpp
#include "SimplifyPath.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SimplifyPath, Solution71_1) {
  auto S71_1 = Solution71_1();

  EXPECT_EQ("/home", S71_1.simplifyPath("/home/"));
  EXPECT_EQ("/", S71_1.simplifyPath("/../"));
  EXPECT_EQ("/home/foo", S71_1.simplifyPath("/home//foo/"));
}
